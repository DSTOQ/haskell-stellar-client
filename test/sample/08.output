TransactionEnvelope
  { _transaction =
      Transaction
        { _sourceAccount =
            PublicKeyEd25519
              { _publicKeyEd25519 =
                  16075459131400611646815471232531135251476096185659753758724518583293378337834
              }
        , _fee = Fee { _fee = 500 }
        , _seqNum = SequenceNumber { _sequenceNumber = 46283138812346447 }
        , _timeBounds = Nothing
        , _memo = MemoNone
        , _operations =
            [ Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling =
                            AssetCreditAlphanum4
                              AssetCode4 { _assetCode4 = 1129208064 }
                              PublicKeyEd25519
                                { _publicKeyEd25519 =
                                    12020110078372443445778830331393468099140010802731689699023048901958341411043
                                }
                        , _buying = AssetNative
                        , _amount = 10000000
                        , _price = Price { _numerator = 7717 , _denominator = 10000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling =
                            AssetCreditAlphanum4
                              AssetCode4 { _assetCode4 = 1431520256 }
                              PublicKeyEd25519
                                { _publicKeyEd25519 =
                                    12020110078372443445778830331393468099140010802731689699023048901958341411043
                                }
                        , _buying = AssetNative
                        , _amount = 100000000
                        , _price = Price { _numerator = 4981 , _denominator = 5000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling = AssetNative
                        , _buying =
                            AssetCreditAlphanum4
                              AssetCode4 { _assetCode4 = 1481789440 }
                              PublicKeyEd25519
                                { _publicKeyEd25519 =
                                    12020110078372443445778830331393468099140010802731689699023048901958341411043
                                }
                        , _amount = 10000000
                        , _price = Price { _numerator = 2073 , _denominator = 5000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling =
                            AssetCreditAlphanum4
                              AssetCode4 { _assetCode4 = 1482245888 }
                              PublicKeyEd25519
                                { _publicKeyEd25519 =
                                    12020110078372443445778830331393468099140010802731689699023048901958341411043
                                }
                        , _buying = AssetNative
                        , _amount = 90000000
                        , _price = Price { _numerator = 4093 , _denominator = 10000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling =
                            AssetCreditAlphanum4
                              AssetCode4 { _assetCode4 = 1481395456 }
                              PublicKeyEd25519
                                { _publicKeyEd25519 =
                                    12020110078372443445778830331393468099140010802731689699023048901958341411043
                                }
                        , _buying = AssetNative
                        , _amount = 90000000
                        , _price = Price { _numerator = 1091 , _denominator = 1250 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            ]
        }
  , _signatures =
      [ DecoratedSignature
          { _hint = SignatureHint "99d49c2a"
          , _signature =
              Signature
                "966ae58983e5d19553a963af63d609ac1a75bcb2eb42fb0f94feb1f3f7c12f044a9ab549dd657d56cb96a55465e460ca68691d49c598ab3bd215d3e99e761e01"
          }
      ]
  }