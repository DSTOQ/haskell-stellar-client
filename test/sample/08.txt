AAAAACOKYy7aaFCAH4+B5u3p0sVdwhHF7m/Mu34VP8CZ1JwqAAAB9ACkbkUAAABPAAAAAAAAAAAAAAAFAAAAAAAAAAMAAAABQ05ZAAAAAAAakyQmWKz4GOEcugg3WE9HV5IlQadnfpkrISF2oZ7A4wAAAAAAAAAAAJiWgAAAHiUAACcQAAAAAAAAAAAAAAAAAAAAAwAAAAFVU0QAAAAAABqTJCZYrPgY4Ry6CDdYT0dXkiVBp2d+mSshIXahnsDjAAAAAAAAAAAF9eEAAAATdQAAE4gAAAAAAAAAAAAAAAAAAAADAAAAAAAAAAFYUlAAAAAAABqTJCZYrPgY4Ry6CDdYT0dXkiVBp2d+mSshIXahnsDjAAAAAACYloAAAAgZAAATiAAAAAAAAAAAAAAAAAAAAAMAAAABWFlHAAAAAAAakyQmWKz4GOEcugg3WE9HV5IlQadnfpkrISF2oZ7A4wAAAAAAAAAABV1KgAAAD/0AACcQAAAAAAAAAAAAAAAAAAAAAwAAAAFYTE0AAAAAABqTJCZYrPgY4Ry6CDdYT0dXkiVBp2d+mSshIXahnsDjAAAAAAAAAAAFXUqAAAAEQwAABOIAAAAAAAAAAAAAAAAAAAABmdScKgAAAECWauWJg+XRlVOpY69j1gmsGnW8sutC+w+U/rHz98EvBEqatUndZX1Wy5alVGXkYMpoaR1JxZirO9IV0+medh4B
