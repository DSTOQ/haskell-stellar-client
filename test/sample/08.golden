TransactionEnvelope
  { _transaction =
      Transaction
        { _sourceAccount =
            GARYUYZO3JUFBAA7R6A6N3PJ2LCV3QQRYXXG7TF3PYKT7QEZ2SOCV2QY
        , _fee = FeeStroops { _feeStroops = 500 }
        , _seqNum = 46283138812346447
        , _timeBounds = Nothing
        , _memo = MemoNone
        , _operations =
            [ Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling =
                            AssetCreditAlphanum
                              CNY GANJGJBGLCWPQGHBDS5AQN2YJ5DVPERFIGTWO7UZFMQSC5VBT3AOGH6P
                        , _buying = AssetNative
                        , _amount = NonNegativeInt64 (Refined 10000000)
                        , _price = Price { _numerator = 7717 , _denominator = 10000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling =
                            AssetCreditAlphanum
                              USD GANJGJBGLCWPQGHBDS5AQN2YJ5DVPERFIGTWO7UZFMQSC5VBT3AOGH6P
                        , _buying = AssetNative
                        , _amount = NonNegativeInt64 (Refined 100000000)
                        , _price = Price { _numerator = 4981 , _denominator = 5000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling = AssetNative
                        , _buying =
                            AssetCreditAlphanum
                              XRP GANJGJBGLCWPQGHBDS5AQN2YJ5DVPERFIGTWO7UZFMQSC5VBT3AOGH6P
                        , _amount = NonNegativeInt64 (Refined 10000000)
                        , _price = Price { _numerator = 2073 , _denominator = 5000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling =
                            AssetCreditAlphanum
                              XYG GANJGJBGLCWPQGHBDS5AQN2YJ5DVPERFIGTWO7UZFMQSC5VBT3AOGH6P
                        , _buying = AssetNative
                        , _amount = NonNegativeInt64 (Refined 90000000)
                        , _price = Price { _numerator = 4093 , _denominator = 10000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling =
                            AssetCreditAlphanum
                              XLM GANJGJBGLCWPQGHBDS5AQN2YJ5DVPERFIGTWO7UZFMQSC5VBT3AOGH6P
                        , _buying = AssetNative
                        , _amount = NonNegativeInt64 (Refined 90000000)
                        , _price = Price { _numerator = 1091 , _denominator = 1250 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            ]
        }
  , _signatures =
      [ DecoratedSignature
          { _hint = SignatureHint { _signatureHint = "99d49c2a" }
          , _signature =
              Signature
                { _signature =
                    "966ae58983e5d19553a963af63d609ac1a75bcb2eb42fb0f94feb1f3f7c12f044a9ab549dd657d56cb96a55465e460ca68691d49c598ab3bd215d3e99e761e01"
                }
          }
      ]
  }